ntbs-ui-tests/Steps/AssertSteps.cs

```diff
@@ -47,9 +50,17 @@ public void ThenIShouldBeOnTheHomepage()
         [Then(@"I should see the Notification")]
         public void ThenIShouldSeeTheNotification()
         {
-            var urlRegex = new Regex(@".*/Notifications/(\d+)/?(#.+)?$");
-            var match = urlRegex.Match(Browser.Url);
-            Assert.True(match.Success, $"Url I am on instead: {Browser.Url}");
+            try
+            {
+                var wait = new WebDriverWait(Browser, Settings.ImplicitWait);
+                wait.Until(ExpectedConditions.UrlMatches(@".*/Notifications/(\d+)/?(#.+)?$"));
+            }
+            catch (WebDriverTimeoutException) {}
+            finally{
+                var urlRegex = new Regex(@".*/Notifications/(\d+)/?(#.+)?$");
+                var match = urlRegex.Match(Browser.Url);
+                Assert.True(match.Success, $"Url I am on instead: {Browser.Url}");
```

Comment:
Isn't this duplicating the wait above? Do we need the try-catch-finally at all here?

Reply:
So it kind of is duplicating the wait, but for a longer time. Some of the times this is asserted there was a preceding "Then I wait" but that only runs for 3s, which was not a long enough wait. Either we could change "And I wait" to wait for a much longer period of time (which will slow the tests down), or we could make use of an polling-based wait, like we use `ImplicitWait` when selecting DOM elements by ID.

I went for the polling-wait approach, using a `WebDriverWait` to keep testing whether the browser URL as changed to the right value. The method returns when the URL becomes the expected one, or throws an exception if it times out. However
the exception it throws doesn't give a particularly good explanation. Instead I caught it, and then would use the nicer Xunit `Assert`.

As per our chat, this is now done on `catch`, rather than `finally`

Reply:
Although, potentially we could use `Assert.Matches` for a better error message